lib/router.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+const express = require('express');
+const bodyParser = require('body-parser');
+const url = require('url');
+//connect to mongoose and open a connection to the test database on local MongoDB
+
+const Employee = require('../model/Employee');
+const router = express.Router();
+
+router.use(bodyParser.json());
+
+//POST
+router.post('/', (req, res, next) => {
+
+  var postedEmployee = req.body;
+  console.log(postedEmployee);
+
+  if (!Object.keys(postedEmployee).length) return res.sendStatus(400);
+
+  var newEmployee = new Employee({
+    _id: postedEmployee._id,
+    name: postedEmployee.name,
+    username: postedEmployee.username,
+    DOB: new Date(postedEmployee.DOB),
+    email: postedEmployee.email,
+    address: postedEmployee.address,
+    phone: postedEmployee.phone,
+    position: postedEmployee.position
+  });
+
+  newEmployee.save((err, savedEmployee) => {
+    if (err) {
+      console.error(err);
+      return res.sendStatus(400);
+    }
+    res.send(savedEmployee);
+  });
+
+});
+
+//General GET
+router.get('/', (req, res, next) => {
+  console.log(url.parse(req.url));
+  Employee.find({}).lean().exec( (err, employees) => {
+    if (err) {
+      console.error(err);
+      res.end();
+    }
+    res.send(employees);
+  });
+
+});
+
+//Specific get
+router.get('/:id', (req, res, next) => {
+  var id = req.params.id;
+  Employee.findById(id, (err, employee) => {
+    if (err) {
+      console.error(err);
+      return res.sendStatus(500);
+    }
+    res.send(employee);
+  });
+});
+
+//PUT
+router.put('/:id', (req, res, next) => {
+  var id = req.params.id;
+  var update = req.body;
+  update.DOB = new Date(update.DOB);
+
+  Employee.findByIdAndUpdate(id, update, {runValidators: true, multi: false}, (err, numAffected) => {
+    if (err) {
+      res.sendStatus(400);
+      return console.error(err);
+    }
+
+    Employee.findById(req.params.id, (err, employee) => {
+      if (err) {
+        console.error(err);
+        return res.sendStatus(500);
+      }
+      res.send(employee);
+    });
+
+  });
+});
+
+//PATCH (Same logic as PUT)
+router.patch('/:id', (req, res, next) => {
```

Comment:
So a patch is different than a put in that the put is supplying __all__ fields to be updated, and if they are absent the value should be cleared. Many examples online however will show the put as you have it, being supplemental to existing data.